common/persistence/namespace_replication_queue.go

```diff
@@ -245,14 +245,10 @@ func (q *namespaceReplicationQueueImpl) updateAckLevel(
 	}
 
 	// Ignore possibly delayed message
-	if ack, ok := ackLevels[clusterName]; ok && ack > lastProcessedMessageID {
+	if ack, ok := ackLevels[clusterName]; ok && ack >= lastProcessedMessageID {
 		return nil
 	}
 
-	// TODO remove this block in 1.12.x
-	delete(ackLevels, "")
```

Comment:
we always have clusterName specified now right? 
This deletion is for deprecated logic?

Reply:
I just see remove this in 1.12 and we are far after 1.12. I will double check we have all places using cluster name.